nifi-manifest/nifi-runtime-manifest-core/src/main/java/org/apache/nifi/runtime/manifest/impl/StandardComponentManifestBuilder.java

```diff
@@ -61,6 +65,24 @@ public ComponentManifestBuilder addReportingTask(final ReportingTaskDefinition r
         return this;
     }
 
+    @Override
+    public ComponentManifestBuilder addParameterProvider(ParameterProviderDefinition parameterProviderDefinition) {
+        if (parameterProviderDefinition == null) {
+            throw new IllegalArgumentException("Parameter Provider definition cannot be null");
+        }
+        parameterProviders.add(parameterProviderDefinition);
+        return this;
+    }
+
+    @Override
+    public ComponentManifestBuilder addFlowAnalysisRule(FlowAnalysisRuleDefinition flowAnalysisRuleDefinition) {
+        if (flowAnalysisRuleDefinition == null) {
+            throw new IllegalArgumentException("Flow Analysis Rule definition cannot be null");
+        }
+        flowAnalysisRules.add(flowAnalysisRuleDefinition);
+        return this;
+    }
+
```

Comment:
There is still one change needed here in the final `build` method, it needs to set the parameter providers and flow analysis rules from the builder into the final `ComponentManifest`.

Reply:
Good catch, done, thanks @bbende